### DISCUSSION TITLE
Frequently hitting HTTPClientError.deadlineExceeded. What's the default deadline? Is it possible to increase the time?

### DISCUSSION THREAD

**Discussion by srazzell** [soto-project/soto#622] [Q&A] [answered]
My team has an app that's often uploading large files to AWS. We are seeing a failure rate somewhere around 10% with this error: "HTTPClientError.deadlineExceeded". I'm assuming it's due to us uploading large files and hitting a timeout/deadline. Is there a default for how long a client has to finish an upload? If so, can it be changed?

**Comment by adam-fowler** [comment, marked as answer]
You can set the timeout in the initialiser for your service object eg
```swift
let s3 = S3(client: AWSClient, region: .useast1, timeout: .seconds(60))
```
You also use an existing service object and create a new from it with the extended timeout
```swift
let s3WithTimeout = s3.with(timeout: .minutes(2))
```

**Reply by srazzell** [reply]
Thank you!
